### PR TITLE
Security page

### PR DESCRIPTION
**Pull request by ardier16**
- [x] Security page
- [x] Password change (+ 2FA)
- [x] 2FA enable/disable
- [x] Account ID & secret seed views
- [x] 2FA for login

### CODE REVIEW

File: src/vue/forms/ChangePasswordForm.vue [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+  methods: {
+    ...mapActions({
+      storeWallet: vuexTypes.STORE_WALLET,
+    }),
+    async submit () {
+      if (!this.isFormValid()) {
+        return
+      }
+      this.disableForm()
+      try {
```
Comment (max-nes) [inline src/vue/forms/ChangePasswordForm.vue]: @quinn71, here goes absolute try-catch hell. not sure we got time to refactor it now, so I created a task for it:
https://app.asana.com/0/894507180698157/1108030165482984/f

Yeah refactor tasks after refactor...

File: src/vue/forms/TfaForm.vue [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    class="app-form tfa-form"
+    @submit.prevent="isFormValid() && enableTfa()"
+  >
+    <template v-if="!form.factor">
+      <div class="app__form-row">
+        <div class="app__form-field">
+          <input-field
+            v-model="form.password"
+            @blur="touchField('form.password')"
+            id="tfa-password"
```
Comment (max-nes) [inline src/vue/forms/TfaForm.vue]: why do we need an id here?

File: src/vue/forms/TfaForm.vue [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+        :label="'tfa-form.secret-lbl' | globalize"
+      />
+      <p class="tfa-form__hint tfa-form__code-hint">
+        {{ 'tfa-form.tfa-code-desc' | globalize }}
+      </p>
+      <div class="app__form-field">
+        <input-field
+          v-model="form.code"
+          @blur="touchField('form.code')"
+          id="tfa-code"
```
Comment (max-nes) [inline src/vue/forms/TfaForm.vue]: why do we need an id here?

File: src/vue/forms/ChangePasswordForm.vue [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+    </div>
+    <div
+      v-if="isTotpEnabled"
+      class="app__form-row"
+    >
+      <div class="app__form-field">
+        <input-field
+          v-model="form.tfaCode"
+          @blur="touchField('form.tfaCode')"
+          id="change-password-tfa-code"
```
Comment (max-nes) [inline src/vue/forms/ChangePasswordForm.vue]: why do we need an id here?

File: src/vue/forms/LoginForm.vue [hunk 5]
```diff
@@ -25,6 +25,21 @@
...
     </div>
+    <div
+      v-if="tfaError"
+      class="app__form-row"
+    >
+      <div class="app__form-field">
+        <input-field
+          v-model="form.tfaCode"
+          @blur="touchField('form.tfaCode')"
+          id="login-tfa-code"
```
Comment (max-nes) [inline src/vue/forms/LoginForm.vue]: why do we need an id here?

File: src/vue/forms/TfaForm.vue [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+  name: 'tfa-form',
+  components: {
+    KeyViewer,
+    ClipboardField,
+  },
+  mixins: [FormMixin],
+  data: _ => ({
+    form: {
+      password: '',
+      factor: null,
```
Comment (max-nes) [inline src/vue/forms/TfaForm.vue]: it is not part of the form. and do not store `null`s as placeholders of objects

File: src/vue/forms/TfaForm.vue [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+      this.disableForm()
+      if (this.isTotpEnabled) {
+        await this.deleteTotpFactor()
+      } else {
+        await this.createTotpFactor()
+      }
+      this.enableForm()
+    },
+    async createTotpFactor () {
+      try {
```
Comment (max-nes) [inline src/vue/forms/TfaForm.vue]: @quinn71, again the try-catch hell. created the task: https://app.asana.com/0/894507180698157/1108030165482985/f

File: src/vue/pages/Security.vue [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+            :label="'security-page.secret-seed-title' | globalize"
+          />
+        </div>
+      </template>
+    </drawer>
+    <div class="security-page__row">
+      <p class="security-page__row-title">
+        {{ 'security-page.enable-tfa-title' | globalize }}
+      </p>
+      <div @click="showDrawer(VIEW_MODES.enableTfa)">
```
Comment (max-nes) [inline src/vue/pages/Security.vue]: do not use @click on non-tabable elements. only buttons and links allowed

File: src/vue/pages/Security.vue [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+        <switch-field :is-enabled="isTotpEnabled" />
```
Comment (max-nes) [inline src/vue/pages/Security.vue]: in the current implementation there can be an issue of the event propagation on some browsers

File: src/vue/pages/Security.vue [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+  .security-page__row {
+    padding: 2.4rem;
+    height: 7.4rem;
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+
+    .security-page__row-title {
+      font-size: 1.8rem;
+      color: $col-primary;
```
Comment (max-nes) [inline src/vue/pages/Security.vue]: do not use $col-primary outside the variables.scss

File: src/vue/pages/Security.vue [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+    }
+
+    .security-page__row-action {
+      font-size: 1.3rem;
+      cursor: pointer;
+      color: $col-primary-lighten;
```
Comment (max-nes) [inline src/vue/pages/Security.vue]: do not use $col-primary-lighten outside the variables.scss

File: src/vue/pages/Security.vue [hunk 12]
```diff
@@ -0,0 +1,202 @@
...
+  hr {
+    margin: 0 2.4rem;
+    border: $col-block-line solid .05rem;
+  }
+}
+
+.secret-seed__description {
+  font-size: 1.2rem;
+  line-height: 1.25;
+  color: $col-secondary;
```
Comment (max-nes) [inline src/vue/pages/Security.vue]: do not use $col-secondary outside the variables.scss

File: src/vue/pages/Security.vue [hunk 13]
```diff
@@ -0,0 +1,202 @@
+<template>
+  <div class="security-page">
```
Comment (max-nes) [inline src/vue/pages/Security.vue]: pls introduce newlines here. it is impossible to navigate over the file

File: src/vue/forms/TfaForm.vue (hunk 2 above)
Comment (ardier16) [reply src/vue/forms/TfaForm.vue]: For the field bindings in unit tests

https://github.com/tokend/web-client/blob/86707d8eea73a3b56e918e00b1d8fc2c374e59be/src/vue/forms/LoginForm.vue#L8 https://github.com/tokend/web-client/blob/86707d8eea73a3b56e918e00b1d8fc2c374e59be/src/vue/forms/LoginForm.vue#L20

https://github.com/tokend/web-client/blob/86707d8eea73a3b56e918e00b1d8fc2c374e59be/src/vue/forms/LoginForm.spec.js#L57-L60

File: src/vue/forms/TfaForm.vue (hunk 3 above)
Comment (ardier16) [reply src/vue/forms/TfaForm.vue]: https://github.com/tokend/web-client/pull/38#discussion_r253804830

File: src/vue/forms/ChangePasswordForm.vue (hunk 4 above)
Comment (ardier16) [reply src/vue/forms/ChangePasswordForm.vue]: https://github.com/tokend/web-client/pull/38#discussion_r253804830

File: src/vue/forms/LoginForm.vue (hunk 5 above)
Comment (ardier16) [reply src/vue/forms/LoginForm.vue]: https://github.com/tokend/web-client/pull/38#discussion_r253804830

File: src/vue/fields/SwitchField.vue [hunk 14]
```diff
@@ -0,0 +1,95 @@
+<template>
+  <div class="switch-field">
+    <button
+      class="switch-field__switch"
```
[the remark this answers is not in the document]
Comment (ardier16) [reply src/vue/fields/SwitchField.vue]: Renamed to `switch-field__toggle`

File: src/vue/fields/SwitchField.vue [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+      @click="toggle"
+    >
+      <span
+        class="switch-field__switch-slider"
```
[the remark this answers is not in the document]
Comment (ardier16) [reply src/vue/fields/SwitchField.vue]: Reanmed to `switch-field__toggle-slider`

File: src/vue/fields/SwitchField.vue [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+      border: solid .1rem darken($col-switch-field-enabled, 10%);
+      transform: translateX(2.5rem);
+    }
+  }
+}
+
+.switch-field__label {
+  margin-left: 1rem;
+  font-size: 1.6rem;
+  color: $col-primary;
```
[the remark this answers is not in the document]
Comment (ardier16) [reply src/vue/fields/SwitchField.vue]: Replaced

File: src/vue/pages/Security.vue (hunk 8 above)
Comment (ardier16) [reply src/vue/pages/Security.vue]: Fixed, use button

File: src/vue/pages/Security.vue (hunk 9 above)
Comment (ardier16) [reply src/vue/pages/Security.vue]: Fixed, replaced by `v-model`

File: src/vue/fields/SwitchField.vue [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+  align-items: center;
+}
+
+.switch-field__switch {
+  position: relative;
+  display: inline-block;
+  width: 5.3rem;
+  height: 2.8rem;
+
+  &-slider {
```
[the remark this answers is not in the document]
Comment (ardier16) [reply src/vue/fields/SwitchField.vue]: Updated, span removed, used `:before` for the button

File: src/vue/fields/SwitchField.vue [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+
+    &:before {
+      position: absolute;
+      content: "";
+      height: 2.2rem;
+      width: 2.2rem;
+      left: .1rem;
+      bottom: .1rem;
+      background-color: $col-switch-field-disabled;
+      border: solid .1rem darken($col-switch-field-disabled, 10%);
```
[the remark this answers is not in the document]
Comment (ardier16) [reply src/vue/fields/SwitchField.vue]: Moved to the `variables.scss`

File: src/vue/forms/TfaForm.vue [hunk 19]
```diff
@@ -135,7 +137,10 @@ export default {
       loadFactors: vuexTypes.LOAD_FACTORS,
     }),
     async toggleFactor () {
-      if (this.$v.form.password.$invalid) return
+      if (this.$v.form.password.$invalid) {
+        this.$v.form.password.$touch()
```
Comment (max-nes) [inline src/vue/forms/TfaForm.vue]: is not is should be done before the `this.$v.form.password.$invalid` check?

Comment (ardier16) [reply src/vue/forms/TfaForm.vue]: Updated, `isFormValid` used

**Pull request merged**